mitodl/open-discussions
# Field Pages for InfiniteCorridor - backend

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of #3584 

#### What's this PR do?
- Adds backend code for `Field Channel` and `Field Moderator` API CRUD.

#### How should this be manually tested?
- As an admin, go to `http://localhost:8063/api/v0/fields/`, you should see a POST form.  Enter a name and title, then submit, you should get a valid response.
- Refresh the url, you should see your new field channel listed, add 1 or 2 more if you like.
- Go to `http://localhost:8063/api/v0/fields/<field_name>/`, you should see details about your field and a POST form.  Try using the form to change title, it should work.
- Go to `http://localhost:8063/api/v0/fields/<field_name>/moderators/`, you should see a POST form.
- Enter either a username or email of another `User` and submit.  You should get a valid response.  Refresh the url and that user should be listed.
- Go to django admin and hijack that user
- Go to `http://localhost:8063/api/v0/fields/`, you should see the list of field channels, but there should not be a POST form (moderators shouldn't have create permissions).
- Go to `http://localhost:8063/api/v0/fields/<field_name>/`, you should see details about your field channel and a POST form.  Try using the form to change the title, it should work.
- Go to `http://localhost:8063/api/v0/fields/<field_name>/moderators/`, you should see a POST form.  Try adding another user, it should work.
- Go to `http://localhost:8063/api/v0/fields/<field_name>/moderators/<username>`, for the user you just added.  Click the `DELETE" button at the top, it should work.

If you have reddit set up locally, you could also try created a new discussion channel (http://localhost:8063/manage/c/new) and then adding posts, comments, moderators, etc, to make sure everything still works.
